app/app_helper.rb

```diff
@@ -0,0 +1,22 @@
+def current_user
+  @current_user ||= User.get(session[:user_id])
+end
+
+def find_user(username_or_email)
+  user = User.first(username: username_or_email)
+  user = User.first(email: username_or_email) if user.nil?
```

Comment:
I think conditional clauses at the end of line statements are a bit abused in ruby, the above code is harder to read because when I first see it I think 'what, he's searching the user twice', then I scan right and I realise 'oh, right, only if the first user is nil', but it's very easy not to scan right and if I wasn't paying special attention to the code I might miss this.

I would prefer to see something like
```
user = User.first(username: username_or_email) || User.first(email: username_or_email) 
```
This code is equivalent to your lines 6 and 7, but makes the intent clearer.